plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -602,8 +602,8 @@ public void testMultipleEqualityDeletesWithNestedFields()
                     equalityFieldIds);
         }
 
-        // TODO: support read equality deletes with nested fields(https://github.com/trinodb/trino/issues/18625)
-        assertThat(query("SELECT * FROM " + tableName)).failure().hasMessageContaining("Multiple entries with same key");
+        assertThat(query("SELECT * FROM " + tableName))
```

Comment:
I don't know whether it is relevant

https://github.com/trinodb/trino/blob/4971ef1a6b07014354d28b22530f1101f482a4fd/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java#L297-L299

I've added the following assertion to see what happens when the column relevant in the equality filter is not present in the `SELECT` statement. Works as expected.
If you add this assertion, pls consider adding a corresponding comment  - worth having there to be protected against regressions 

```
        // verify that the equality delete is effective also when not specifying the corresponding column in the projection list
        assertThat(query("SELECT id FROM " + tableName))
                .matches("VALUES (BIGINT '1')");
```